### ISSUE TITLE
Does it still work?

### ISSUE THREAD

**Issue by zxl634** [martinfinke/xcall#10]
I've happily used this for years, but it seems to have stopped working (for me at least) recently. Testing it with `/Applications/xcall.app/Contents/MacOS/xcall -url "reminders://x-callback-url/create?title=TestReminder"`, but it simply seems to stall. Have tried to look in Console to look for clues, but my knowledge of MacOS apps and how to debug them is limited.

**Comment by kimaldis** [comment]
Same here. Last release was a while back, I guess it might be an OS upgrade thing. Shame if this is the end of it.

**Comment by kimaldis** [comment]
OK, so I rebuilt it under sonoma for MacOs 10.12 and now it's just fine.

**Comment by gusutabopb** [comment]
Same issue here. I have absolutely no experience with macOS/iOS app dev. I managed to rebuild the app under macOS 14.1, but the binary still does not work.

**Comment by kimaldis** [comment]
As far as I can see this is still broken and the rebuild solution I mentioned above now longer works for me. Is there any real chance of getting it fixed, or if not, is there an alternative anywere? It's a really useful tool.

**Comment by ks3** [comment]
I'm able to run the prebuilt app without any issues on an up to date MacBook Pro with M2 chip.

**Comment by kimaldis** [comment]
Does it handle x-success, though? For example, does `bear://x-callback-url/open-note?title=foo` return note information to the caller?

**Comment by ks3** [comment]
Yep, I'm using it in a script - https://github.com/ks3/cub - to view and edit notes from Bear.

**Comment by kimaldis** [comment]
Interesting. FOr me, the following brings vear up with my note 'Scratch', nothing printed to the terminal and xcall hangs.

`./xcall.app/Contents/MacOS/xcall -url 'bear://x-callback-url/open-note?title=Scratch'`

**Comment by kimaldis** [comment]
Also, the file date on the version I'm using, downloaded using the Latest Version link, is 2017, 5 years prior to the pull requests that highlight the isssue. I'm uding an M1 Macbook pro running Sonoma

**Comment by kimaldis** [comment]
I'd be happy to build a version with the necessary fixes but I'm ignorant of most things Github so I'd need a bit of guidance on how to incorporate them into the existing code.

**Comment by ks3** [comment]
Weird. I assume I'm using the same version as you. Here's some terminal output in case it's helpful. This is from Sonoma 14.4.1 on an M2 Max.

```
$ ls -l xcall.app/Contents/MacOS/xcall
-rwxr-xr-x@ 1 kevin  admin  49520 Apr 19  2017 xcall.app/Contents/MacOS/xcall*
$ sha256sum xcall.app/Contents/MacOS/xcall
822c1a7113fd19e058c5fb0327ca816ace6dc648863ce68cdb529813d6c31f94  xcall.app/Contents/MacOS/xcall
$ ./xcall.app/Contents/MacOS/xcall -url 'bear://x-callback-url/open-note?title=Scratch'
{
  "" : null,
  "note" : "# Scratch\n\nTest note. Velit veniam ea dolor voluptate sed lorem. Ea dolor voluptate sed lorem dolore dolor. Voluptate sed lorem, dolore dolor aute nulla.",
  "modificationDate" : "2024-03-29T14:10:01Z",
  "creationDate" : "2024-03-29T14:08:44Z",
  "title" : "Scratch",
  "is_trashed" : "no",
  "identifier" : "DC4B38B8-FF28-47F8-8B3C-62A335B22996",
  "tags" : "[]"
}
```
